### PR TITLE
feat(dependency dashboard): add option to open all prs

### LINKED ISSUE #15912: Dependency dashboard - add option to open all "Pending Approval" PRs and "Rate limited" PRs
### What would you like Renovate to be able to do?

In 2 of the dependency dashboard sectors - "Pending Approval" and "Rate limited", add an option to manually open all pending PRs. a small feature to add to save some manual work to the user

1. add checkbox - "Approve all pending PRs"
note - will be top checkbox - differentiate then rest of PR checkboxes
image

2. add checkbox - "Approve opening all rate-limited PRs"
note - will be top checkbox - differentiate then rest of PR checkboxes
image

### If you have any ideas on how this should be implemented, please tell us here.

no

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by MaronHatoum**
## Changes
added checkbox - "Approve all pending PRs"
added checkbox - "Open all rate-limited PRs"
I implemented this feature by using the existing functionality, if the new checkbox is selected the new code will collect all the relevant branches (approve-branch/unlimit-branch) and add them to `dependencyDashboardChecks` during runtime.

## Context
Closes https://github.com/renovatebot/renovate/issues/15912

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/dependency-dashboard.ts [hunk 1]
```diff
@@ -15,27 +15,77 @@ import { PackageFiles } from './package-files';
...
+): Record<string, string> {
+  const generalCheckMatch = ' <!-- ([a-zA-Z]+)-branch=([^\\s]+) -->';
+  const allRelevantBranches = selectAllRelevantBranches(issueBody);
+  const re = regEx(generalCheckMatch);
+  allRelevantBranches.forEach((branch) => {
+    const [, type, branchName] = re.exec(branch)!;
+    dependencyDashboardChecks[branchName] = type;
+  });
+  return dependencyDashboardChecks;
+}
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:63]: I don't think this is the right way here. This should be checked per branch in `update/branch/index` below, like we do for the other checkboxes

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt [hunk 2]
```diff
@@ -4,6 +4,7 @@ This issue lists Renovate updates and detected dependencies. Read the [Dependenc
 
 These branches will be created by Renovate only once you click their checkbox below.
 
+ - [ ] <!-- approve-all-pending-prs -->**Approve all pending PRs**
```
Comment (HonkingGoose) [inline lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: These buttons are different because they open _all_ PRs in their section. This also means the button should _look_ different, like one of these maybe:

- [ ] 💯 **Approve all pending PRs** 💯 
- [ ] :rotating_light: **Approve all pending PRs** :rotating_light:
- [ ] :warning: **Approve all pending PRs** :warning:

---

Also, can we use emojis on all of our supported platforms?

- GitHub (.com and Enterprise Server)
- GitLab (.com and CE/EE)
- Bitbucket Cloud
- Bitbucket Server
- Azure DevOps
- Gitea

File: lib/workers/repository/dependency-dashboard.ts (hunk 1 above)
Comment (MaronHatoum) [reply lib/workers/repository/dependency-dashboard.ts:63]: The logic I added here once the user marks (`open-all-rate-limited-prs` or `approve-all-pending-prs`) all the relevant branches (`approve-branch` or `unlimit-branch`) will be added to `dependencyDashboardChecks` and this is the same logic we have today for all the marked branches.

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt (hunk 2 above)
Comment (MaronHatoum) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: @betterPeleg

Comment (betterPeleg) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: ❗

Comment (betterPeleg) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: let's go with this type of style - 
 ⚠️ **Approve all pending PRs** ⚠️

Comment (MaronHatoum) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: what about **Open all rate-limited PRs** should I add ⚠️ also?

Comment (betterPeleg) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: yes. same

Comment (MaronHatoum) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: after your suggestion this is the final result:
https://github.com/MaronHatoum/select-all/issues/68
image

- is this what we expect the dashboard to look like?

Comment (viceice) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: I think those all boxes should be last instead of first?

@rarkins WDYT?

Comment (betterPeleg) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: yeah i agree as well. 
we just thought it might be an issue if there are a lot of pending PRs.
but i agree that for the common use its better to appear last

Comment (MaronHatoum) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:7]: this is an example of the boxes appear last, but as @betterPeleg said if we have a lot of pending approval or rate limited the user should scroll down to mark the new boxes.
https://github.com/MaronHatoum/select-all/issues/68

image

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt [hunk 3]
```diff
@@ -18,6 +19,7 @@ These updates are awaiting their schedule. Click on a checkbox to get an update
 
 These updates are currently rate limited. Click on a checkbox below to force their creation now.
 
+ - [ ] <!-- open-all-rate-limited-prs -->**Open all rate-limited PRs**
```
Comment (HonkingGoose) [inline lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:22]: We recently had some problems with really large Dependency Dashboards not showing all interactive elements:

- #16373

We may run into a similar problem after we merge this PR. Where users with many dependencies can't use the "give me all the PRs" button.

Is there a way to make sure the `Open all $section-name PRs` buttons are always visible and usable?

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt (hunk 2 above)
Comment (viceice) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt]: the user should review all listed pr's anyways before checking the all box anyways

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt (hunk 3 above)
Comment (betterPeleg) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt]: yes @Gabriel-Ladzaretti was working on it.
Gabi do you think such a thing is an option?

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt [hunk 4]
```diff
@@ -6,6 +6,7 @@ These branches will be created by Renovate only once you click their checkbox be
 
  - [ ] <!-- approve-branch=branchName1 -->pr1
  - [ ] <!-- approve-branch=branchName2 -->pr2
+ - [ ] <!-- approve-all-pending-prs -->⚠ **Approve all pending PRs** ⚠
```
Comment (HonkingGoose) [inline lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:9]: Do we want to label this `breaking` and put it in a new `major` release of Renovate?

My thinking is that this is a big change to the amount of "noise/mess" you can make when clicking a checkbox on the Dependency Dashboard. I think self-hosted Renovate users in large organization would like a proper warning about this change.

What do you maintainers think?

Comment (rarkins) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:9]: I don't think we need to consider it major or warn. There are many other ways already to create noise and the user shouldn't do this by accident

File: lib/workers/repository/dependency-dashboard.ts [hunk 5]
```diff
@@ -15,27 +15,77 @@ import { PackageFiles } from './package-files';
...
 
 function checkRebaseAll(issueBody: string): boolean {
   return issueBody.includes(' - [x] <!-- rebase-all-open-prs -->');
 }
 
+function selectAllRelevantBranches(issueBody: string): string[] {
+  const checkedBranches = [];
+  if (checkOpenAllRateLimitedPR(issueBody)) {
+    const checkMatchRate = ' - \\[ \\] <!-- unlimit-branch=([^\\s]+) -->';
+    const checkedRate = issueBody.match(regEx(checkMatchRate, 'g')) ?? [];
```
Comment (rarkins) [inline lib/workers/repository/dependency-dashboard.ts:38]: Can this regEx be moved up and outside of the function so that it's defined once?

File: lib/workers/repository/dependency-dashboard.ts [hunk 6]
```diff
@@ -15,27 +15,77 @@ import { PackageFiles } from './package-files';
...
+    checkedBranches.push(...checkedRate);
+  }
+  if (checkApproveAllPendingPR(issueBody)) {
+    const checkMatchPendingApproval =
+      ' - \\[ \\] <!-- approve-branch=([^\\s]+) -->';
+    const checkedPending =
+      issueBody.match(regEx(checkMatchPendingApproval, 'g')) ?? [];
```
Comment (rarkins) [inline lib/workers/repository/dependency-dashboard.ts:45]: same here

File: lib/workers/repository/dependency-dashboard.ts [hunk 7]
```diff
@@ -15,27 +15,77 @@ import { PackageFiles } from './package-files';
...
+  return checkedBranches;
+}
+
+function getAllSelectedBranches(
+  issueBody: string,
+  dependencyDashboardChecks: Record<string, string>
+): Record<string, string> {
+  const generalCheckMatch = ' <!-- ([a-zA-Z]+)-branch=([^\\s]+) -->';
+  const allRelevantBranches = selectAllRelevantBranches(issueBody);
+  const re = regEx(generalCheckMatch);
```
Comment (rarkins) [inline lib/workers/repository/dependency-dashboard.ts:57]: same here

**Review by rarkins — DISMISSED** [review]

File: lib/workers/repository/update/branch/index.ts [hunk 8]
```diff
@@ -355,10 +355,19 @@ export async function processBranch(
...
       !!config.rebaseRequested;
-
+    const userApproveAllPendingPR = !!config.dependencyDashboardAllPending;
+    const userOpenAllRateLimtedPR = !!config.dependencyDashboardAllRateLimited;
     if (userRebaseRequested) {
       logger.debug('Manual rebase requested via Dependency Dashboard');
       config.reuseExistingBranch = false;
+    } else if (userApproveAllPendingPR) {
+      logger.debug(
+        'Manual Approve All Pending PRs requested via Dependency Dashboard'
```
Comment (HonkingGoose) [inline lib/workers/repository/update/branch/index.ts:365]: This is hard to parse for me, can we try something like:

```suggestion
        'A user manually approved all pending PRs via the Dependency Dashboard.'
```

File: lib/workers/repository/update/branch/index.ts [hunk 9]
```diff
@@ -355,10 +355,19 @@ export async function processBranch(
...
+      );
+    } else if (userOpenAllRateLimtedPR) {
+      logger.debug(
+        'Manual Open All Rate limited PRs requested via Dependency Dashboard'
```
Comment (HonkingGoose) [inline lib/workers/repository/update/branch/index.ts:369]: Same thing here:

```suggestion
        'A user manually approved all rate-limited PRs via the Dependency Dashboard.'
```

File: lib/workers/repository/update/branch/index.spec.ts [hunk 10]
```diff
@@ -1717,5 +1717,65 @@ describe('workers/repository/update/branch/index', () => {
...
+          dependencyDashboardAllPending: true,
+        })
+      ).toEqual({
+        branchExists: true,
+        prNo: undefined,
+        result: 'done',
+      });
+    });
+
+    it('Dependency Dashboard open all rate limited', async () => {
```
Comment (HonkingGoose) [inline lib/workers/repository/update/branch/index.spec.ts:1751]:
```suggestion
    it('Dependency Dashboard open all rate-limited', async () => {
```

File: lib/workers/repository/dependency-dashboard.spec.ts [hunk 11]
```diff
@@ -96,6 +101,58 @@ describe('workers/repository/dependency-dashboard', () => {
...
+        dependencyDashboardIssue: 1,
+        dependencyDashboardRebaseAllOpen: false,
+        dependencyDashboardTitle: 'Dependency Dashboard',
+        prCreation: 'approval',
+        dependencyDashboardAllPending: true,
+        dependencyDashboardAllRateLimited: false,
+      });
+    });
+
+    it('reads dashboard body open all rate limited', async () => {
```
Comment (HonkingGoose) [inline lib/workers/repository/dependency-dashboard.spec.ts:131]:
```suggestion
    it('reads dashboard body open all rate-limited', async () => {
```

File: lib/workers/repository/dependency-dashboard.spec.ts [hunk 12]
```diff
@@ -522,6 +579,123 @@ describe('workers/repository/dependency-dashboard', () => {
...
+        )
+      ).toBeTrue();
+      expect(
+        checkApprovePendingBranch2.test(
+          platform.ensureIssue.mock.calls[0][0].body
+        )
+      ).toBeTrue();
+    });
+
+    it('dependency Dashboard Open All Rate Limited', async () => {
```
Comment (HonkingGoose) [inline lib/workers/repository/dependency-dashboard.spec.ts:644]:
```suggestion
    it('dependency Dashboard Open All rate-limited', async () => {
```

File: lib/workers/repository/dependency-dashboard.spec.ts [hunk 13]
```diff
@@ -522,6 +579,123 @@ describe('workers/repository/dependency-dashboard', () => {
...
+      config.dependencyDashboardChecks = {
+        branchName1: 'unlimit-branch',
+        branchName2: 'unlimit-branch',
+      };
+      config.dependencyDashboardIssue = 1;
+      jest.spyOn(platform, 'getIssue').mockResolvedValueOnce({
+        title: 'Dependency Dashboard',
+        body: `This issue contains a list of Renovate updates and their statuses.
+        ## Rate Limited
+        These updates are currently rate limited. Click on a checkbox below to force their creation now.
```
Comment (HonkingGoose) [inline lib/workers/repository/dependency-dashboard.spec.ts:671]:
```suggestion
        These updates are currently rate-limited. Click on a checkbox below to force their creation now.
```

File: lib/workers/repository/dependency-dashboard.spec.ts [hunk 14]
```diff
@@ -522,6 +579,123 @@ describe('workers/repository/dependency-dashboard', () => {
...
+      config.dependencyDashboard = true;
+      config.dependencyDashboardChecks = {
+        branchName1: 'unlimit-branch',
+        branchName2: 'unlimit-branch',
+      };
+      config.dependencyDashboardIssue = 1;
+      jest.spyOn(platform, 'getIssue').mockResolvedValueOnce({
+        title: 'Dependency Dashboard',
+        body: `This issue contains a list of Renovate updates and their statuses.
+        ## Rate Limited
```
Comment (HonkingGoose) [inline lib/workers/repository/dependency-dashboard.spec.ts:670]:
```suggestion
        ## Rate-limited
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
We should be consistent and use the word `rate-limited` everywhere. Also the debug messages need some work.

**Comment by rarkins** [comment]
@betterPeleg can you test this manually to make sure it functions according to your expectations?

**Comment by betterPeleg** [comment]
> @betterPeleg can you test this manually to make sure it functions according to your expectations?

I have tested it along with Maron and it seems to function according to the expectations.

I do however suggest some cosmetic changes:
- change text to match the "rebase all open PRs" functionality - "Open all rate-limited PRs" -> "Open all rate-limited PRs at once"
- "Approve all pending PRs" -> "Create all pending approval PRs at once" - clearer description of what will happen
- i would also suggest changing the warning emojis to something less alarming - maybe an "unlock" emoji like that - 
image

@HonkingGoose what do you think?

**Comment by HonkingGoose** [comment]
> * change text to match the "rebase all open PRs" functionality - "Open all rate-limited PRs" -> "Open all rate-limited PRs at once"
> - "Approve all pending PRs" -> "Create all pending approval PRs at once" - clearer description of what will happen

Good catch, I like them both. 😉 

> * i would also suggest changing the warning emojis to something less alarming - maybe an "unlock" emoji like that -

The warning emoji should:

- not match any that we already use in Renovate PR body text: 📅🚦♻️🔕👻 
- clearly say: "this action is different from the others on the dashboard"
- re-assure our users that nothing bad is going to happen after they click on it

I like the idea to use a unlock emoji 🔓. But on macOS the icon looks too much like a closed lock. The gap between the lock and the handle is too small to see.

**Comment by betterPeleg** [comment]
How about a lock and key? 🔐
not perfect but also relates to "unlock"

Screen Shot 2022-08-15 at 17 27 44

based on
https://www.emojiall.com/en/emoji/%F0%9F%94%90

**Comment by HonkingGoose** [comment]
I like the 🔐 emoji best, out of the ones we've discussed. We won't find the _perfect_ emoji anyway. 🙈

**Comment by betterPeleg** [comment]
OK
let's do that then.
@MaronHatoum can you please update the PR?

**Comment by MaronHatoum** [comment]
@betterPeleg @HonkingGoose 
I prepared two examples to see what the dashboard will look like:

**Rate limited:**
https://github.com/MaronHatoum/select-all/issues/90
image

**Pending Approval:** 
 https://github.com/MaronHatoum/select-all/issues/91
image

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt [hunk 15]
```diff
@@ -6,6 +6,7 @@ These branches will be created by Renovate only once you click their checkbox be
 
  - [ ] <!-- approve-branch=branchName1 -->pr1
  - [ ] <!-- approve-branch=branchName2 -->pr2
+ - [ ] <!-- approve-all-pending-prs -->🔐 **Create all pending approval PRs at once** 🔐
```
Comment (HonkingGoose) [inline lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:9]: We should use _one_ word for the same concept. Right now we use both `create` and `open`.

- `Create all ... PRs at once`
- `Open all ... PRs at once`

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt [hunk 16]
```diff
@@ -20,6 +21,7 @@ These updates are currently rate limited. Click on a checkbox below to force the
```
Comment (HonkingGoose) [inline lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:21]: I think it looks better if we use `rate-limited` with the `-` in between the words.
To be consistent we should change:

- the `h2` heading to `## Rate-limited`
- the sentence to `These updates are currently rate-limited.`

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt (hunk 15 above)
Comment (betterPeleg) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:9]: I was thinking to use both "open" and "create" since at the "rate-limited" section, the concept is that the PRs are already open but are just blocked by rate limitation, and at the "pending approval" section it was not yet created since it requires approval first.
If you think it is better to use same wording, I would go with "create" since we use the term creation in the titles of the two.

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt (hunk 16 above)
Comment (betterPeleg) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:21]: agreed

File: lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt (hunk 15 above)
Comment (HonkingGoose) [reply lib/workers/repository/__fixtures__/master-issue_with_8_PR.txt:9]: Yeah that makes sense. I'll let you decide what you like best then. 😉

**Review by HonkingGoose — DISMISSED** [review]

**Comment by MaronHatoum** [comment]
Are any more changes needed from me?

File: lib/workers/repository/dependency-dashboard.ts [hunk 17]
```diff
@@ -16,27 +16,81 @@ import { PackageFiles } from './package-files';
...
 }
 
 function checkRebaseAll(issueBody: string): boolean {
   return issueBody.includes(' - [x] <!-- rebase-all-open-prs -->');
 }
 
+function selectAllRelevantBranches(issueBody: string): string[] {
+  const checkedBranches = [];
+  if (checkOpenAllRateLimitedPR(issueBody)) {
+    const checkedRate = issueBody.match(rateLimitedRe) ?? [];
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:49]: shouldn't we use matchAll instead? 🤔

File: lib/workers/repository/dependency-dashboard.ts [hunk 18]
```diff
@@ -16,27 +16,81 @@ import { PackageFiles } from './package-files';
...
+  }
+  return checkedBranches;
+}
+
+function getAllSelectedBranches(
+  issueBody: string,
+  dependencyDashboardChecks: Record<string, string>
+): Record<string, string> {
+  const allRelevantBranches = selectAllRelevantBranches(issueBody);
+  allRelevantBranches.forEach((branch) => {
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:64]: please use a for of loop instead

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/workers/repository/dependency-dashboard.ts [hunk 19]
```diff
@@ -16,27 +16,85 @@ import { PackageFiles } from './package-files';
...
+const checkMatchRate = ' - \\[ \\] <!-- unlimit-branch=([^\\s]+) -->';
+const rateLimitedRe = regEx(checkMatchRate, 'g');
+const checkMatchPendingApproval =
+  ' - \\[ \\] <!-- approve-branch=([^\\s]+) -->';
+const pendingApprovalRe = regEx(checkMatchPendingApproval, 'g');
+const checkMatchGeneral = ' <!-- ([a-zA-Z]+)-branch=([^\\s]+) -->';
+const generalBranchRe = regEx(checkMatchGeneral);
+const checkMatchMarkedBranches =
+  ' - \\[x\\] <!-- ([a-zA-Z]+)-branch=([^\\s]+) -->';
+const markedBranchesRe = regEx(checkMatchMarkedBranches, 'g');
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:32]:
```suggestion
const rateLimitedRe = regEx(' - \\[ \\] <!-- unlimit-branch=([^\\s]+) -->', 'g');
const pendingApprovalRe = regEx(' - \\[ \\] <!-- approve-branch=([^\\s]+) -->', 'g');
const generalBranchRe = regEx(' <!-- ([a-zA-Z]+)-branch=([^\\s]+) -->');
const markedBranchesRe = regEx(' - \\[x\\] <!-- ([a-zA-Z]+)-branch=([^\\s]+) -->', 'g');
```

inline static strings, makes it more readable. we also don't use the strings anywhre else.

File: lib/workers/repository/dependency-dashboard.ts [hunk 20]
```diff
@@ -16,27 +16,88 @@ import { PackageFiles } from './package-files';
...
+  return issueBody.includes(' - [x] <!-- approve-all-pending-prs -->');
 }
 
 function checkRebaseAll(issueBody: string): boolean {
   return issueBody.includes(' - [x] <!-- rebase-all-open-prs -->');
 }
 
+function selectAllRelevantBranches(issueBody: string): string[] {
+  const checkedBranches = [];
+  if (checkOpenAllRateLimitedPR(issueBody)) {
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:51]: why not pass this value from parent function, so we only need to search once?

File: lib/workers/repository/dependency-dashboard.ts [hunk 21]
```diff
@@ -16,27 +16,88 @@ import { PackageFiles } from './package-files';
...
+    const checkedRate = [...issueBody.matchAll(rateLimitedRe)];
+    for (const match of checkedRate) {
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:53]:
```suggestion
    for (const match of issueBody.matchAll(rateLimitedRe)) {
```

we can inline this, the variable isn't used otherwise

File: lib/workers/repository/dependency-dashboard.ts [hunk 22]
```diff
@@ -16,27 +16,88 @@ import { PackageFiles } from './package-files';
...
+      checkedBranches.push(match[0]);
+    }
+  }
+  if (checkApproveAllPendingPR(issueBody)) {
+    const checkedPending = [...issueBody.matchAll(pendingApprovalRe)];
+    for (const match of checkedPending) {
+      checkedBranches.push(match[0]);
+    }
+  }
```
Comment (viceice) [inline lib/workers/repository/dependency-dashboard.ts:62]: same here

File: lib/workers/repository/dependency-dashboard.ts (hunk 20 above)
Comment (MaronHatoum) [reply lib/workers/repository/dependency-dashboard.ts:51]: because I believe if each line contains one action it will be more readable.
but in this case, it is also readable so I will push the changes.

**Review by viceice — DISMISSED** [review]

**Comment by MaronHatoum** [comment]
changes in unit tests after Rahul's changes.

**Review by viceice — APPROVED** [review]
please prefer `partial<T>({})` instead of ` { } as T` in future tests 😉
